packages/blade/src/utils/useBreakpoint/useBreakpoint.ts

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useState, useCallback, useMemo } from 'react';
+import type { Breakpoints } from '../../tokens/global';
+
+const deviceType = {
+  desktop: 'desktop',
+  mobile: 'mobile',
+} as const;
+
+type DeviceType = keyof typeof deviceType;
+type Breakpoint = keyof Breakpoints | undefined;
+
+type BreakpointAndDevice = {
+  matchedBreakpoint: Breakpoint;
+  matchedDeviceType: DeviceType;
+};
+
+const useBreakpoint = ({ breakpoints }: { breakpoints: Breakpoints }): BreakpointAndDevice => {
+  const supportsMatchMedia =
+    typeof document !== 'undefined' && typeof window?.matchMedia === 'function';
+
+  const breakpointsTokenAndQueryCollection = useMemo(
+    () =>
+      (supportsMatchMedia
+        ? Object.entries(breakpoints).map(
+            ([breakpointTokenName, breakpointSize], index, breakpointsArray) => {
+              let mediaQuery = '';
+
+              if (breakpointTokenName === 'max') {
+                mediaQuery = `screen and (min-width: ${breakpointSize}px)`;
+              } else if (breakpointsArray[index - 1]) {
+                mediaQuery = `screen and (min-width: ${
+                  breakpointsArray[index - 1][1] + 1
+                }px) and (max-width: ${breakpointSize}px)`;
+              } else {
+                mediaQuery = `screen and (max-width: ${breakpointSize}px)`;
+              }
+
+              return [breakpointTokenName, breakpointSize, mediaQuery];
+            },
+          )
+        : []) as [keyof Breakpoints, number, string][],
+    [breakpoints, supportsMatchMedia],
+  );
+
+  const getMatchedDeviceType = useCallback((matchedBreakpoint: Breakpoint): DeviceType => {
+    let matchedDeviceType: DeviceType = deviceType.mobile;
+    if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {
+      // react-native
+      matchedDeviceType = deviceType.mobile;
+    } else if (typeof document !== 'undefined') {
+      // browser
+      if (matchedBreakpoint && ['xs', 's', 'm'].includes(matchedBreakpoint)) {
+        // tablet is also categorised as mobile
+        matchedDeviceType = deviceType.mobile;
+      } else {
+        matchedDeviceType = deviceType.desktop;
+      }
+    } else if (typeof process !== 'undefined') {
+      // node
+      //@TODO: Check for useragent for node
+      matchedDeviceType = deviceType.desktop;
```

Comment:
For ssr here, change this to mobile by default

Reply:
hmm ok for web I'm keeping it to desktop for now since majority of our apps are desktop